src/js/fluxcontroller.js

```diff
@@ -221,7 +221,16 @@ define(function (require, exports, module) {
     FluxController.prototype._initActionLocks = function () {
         var actionDependencies = new Map();
 
-        var _resolveDependencies = function (action) {
+        var _resolveDependencies = function (action, stack) {
+            stack = stack || [];
```

Comment:
I like this change! But I think it would make more sense for this to be a `Set` also? Then you can do a `log n` `has` membership check instead of an `O(n)` `contains` check. Also, we should document the parameters and return type of this function while we're in here.